vast_pipeline/plots.py

```diff
@@ -261,3 +304,289 @@ def plot_lightcurve(
+        use_int_flux: Use integrated fluxes for the analysis instead of
+            peak fluxes, defaults to 'False'.
+
+    Returns: Tuple containing the eta_fit_mean, eta_fit_sigma, v_fit_mean
+        and the v_fit_sigma.
+    """
+
+    if use_peak_flux:
+        eta_label = 'eta_peak'
+        v_label = 'v_peak'
+    else:
+        eta_label = 'eta_int'
+        v_label = 'v_int'
+
+    eta_log = np.log10(df[eta_label])
+    v_log = np.log10(df[v_label])
+
+    eta_log_clipped = sigma_clip(
+        eta_log, masked=False, stdfunc=mad_std, sigma=3
+    )
+    v_log_clipped = sigma_clip(
+        v_log, masked=False, stdfunc=mad_std, sigma=3
+    )
+
+    eta_fit_mean, eta_fit_sigma = norm.fit(eta_log_clipped)
+    v_fit_mean, v_fit_sigma = norm.fit(v_log_clipped)
+
+    return (eta_fit_mean, eta_fit_sigma, v_fit_mean, v_fit_sigma)
+
+
+def gaussian_fit(
+    data: pd.Series, param_mean: float, param_sigma: float
+) -> Tuple[np.ndarray, norm]:
+    """
+    Returns the Guassian to add to the matplotlib plot.
+
+    Args:
+        data: Series object containing the log10 values of the
+            distribution to plot.
+        param_mean: The calculated mean of the Gaussian to fit.
+        param_sigma: The calculated sigma of the Gaussian to fit.
+
+    Returns:
+        Tuple containing the range of the returned data and the
+        Gaussian fit.
+    """
+    range_data = np.linspace(min(data), max(data), 1000)
+    fit = norm.pdf(range_data, loc=param_mean, scale=param_sigma)
+
+    return range_data, fit
+
+
+def make_bins(self, x: pd.Series) -> List[float]:
+    """
+    Calculates the bins that should be used for the v, eta distribution
+    using bayesian blocks.
+
+    Args:
+        x: Series object containing the log10 values of the
+            distribution to plot.
+
+    Returns:
+        Bins to apply.
+    """
+    new_bins = bayesian_blocks(x)
+    binsx = [
+        new_bins[a] for a in range(
+            len(new_bins) - 1
+        ) if abs((new_bins[a + 1] - new_bins[a]) / new_bins[a]) > 0.05
+    ]
+    binsx = binsx + [new_bins[-1]]
+
+    return binsx
+
+
+def plot_eta_v_bokeh(
+    source: Source,
+    eta_sigma: float,
+    v_sigma: float,
+    use_peak_flux: bool = True
+) -> gridplot:
+    """
+    Adapted from code written by Andrew O'Brien.
+    Produces the eta, V candidates plot
+    (see Rowlinson et al., 2018,
+    https://ui.adsabs.harvard.edu/abs/2019A%26C....27..111R/abstract).
+    Returns a bokeh version.
+
+    Args:
+        df: Dataframe containing the sources from the pipeline run. A
+            `pandas.core.frame.DataFrame` instance.
+        eta_sigma: The log10 eta_cutoff from the analysis.
+        v_sigma: The log10 v_cutoff from the analysis.
+        use_peak_flux: Use peak fluxes for the analysis instead of
+            integrated fluxes, defaults to 'True'.
+
+    Returns:
+        Bokeh grid object containing figure.
+    """
+    df = pd.DataFrame(source.values())
```

Comment:
Selecting only the columns we need will help performance

```suggestion
    df = pd.DataFrame(source.values("id", "eta_peak", "eta_int", "v_peak", "v_int", "n_meas_sel"))
```